server/js/app/wireless-debug.js

```diff
@@ -0,0 +1,241 @@
+  }
+
+  /**
+   * Opens a WebSocket connection to the server
+   */
+  start() {
+    this.ws_ = new WebSocket(`ws://${location.host}/ws`);
+    this.ws_.onopen = () => this.websocketOnOpen();
+    this.ws_.onmessage = (message) => this.websocketOnMessage(message);
+
+    let api_key = $('#api-key');
+    let device = $('#device');
+    let app = $('#app');
+    let time = $('#start-time');
+    let data_table = $('#historical-log-table').DataTable();
+
+    device.on('change', () => {
+      let chosen_device = device.val();
+      // Gets rid of old data but keeps table structure.
+      data_table.destroy();
+      if (chosen_device !== 'None') {
+        $('#hidden-dev-alias').css('display', 'block');
+        $('#hidden-app').css('display', 'block');
+        $('#hidden-app-alias').css('display', 'none');
+        $('#dev-alias').val('');
+        $('#hidden-start').css('display', 'none');
+        let data = {
+          'apiKey': api_key.html(),
+          'device': chosen_device,
+        };
+        $.ajax({
+          url: '/appList',
+          data: data,
+          cache: false,
+          success: function(data) {
+            app.empty();
+            time.empty();
+            $('#historical-log-table tbody tr').remove();
+            app.append('<option value="None"></option>');
+            for (let i of data.apps) {
+              app.append(`<option value="${i}">${i}</option>`);
+            }
+          },
+        });
+      } else {
+        $('#hidden-dev-alias').css('display', 'none');
+        $('#hidden-app').css('display', 'none');
+        $('#hidden-start').css('display', 'none');
+        $('#historical-log-table tbody tr').remove();
+      }
+    });
+    app.on('change', () => {
+      let chosen_app = app.val();
+      // Gets rid of old data but keeps table structure.
+      data_table.destroy();
+      if (chosen_app !== 'None') {
+        $('#hidden-app-alias').css('display', 'block');
+        $('#hidden-start').css('display', 'block');
+        $('#app-alias').val('');
+        let data = {
+          'apiKey': api_key.html(),
+          'device': device.val(),
+          'app': chosen_app,
+        };
+        $.ajax({
+          url: '/sessionList',
+          data: data,
+          cache: false,
+          success: function(data) {
+            time.empty();
+            $('#historical-log-table tbody tr').remove();
+            time.append('<option value="None"></option>');
+            for (let i of data.starttimes) {
+              time.append(`<option value="${i}">${i}</option>`);
+            }
+          },
+        });
+      } else {
+        $('#hidden-app-alias').css('display', 'none');
+        $('#hidden-start').css('display', 'none');
+        $('#historical-log-table tbody tr').remove();
+      }
+    });
+
+    time.on('change', () => {
+      let chosen_starttime = time.val();
+      // Gets rid of old data but keeps table structure.
+      data_table.destroy();
```

Comment:
Need to add `data_table = $('#historical-log-table').DataTable();` before the `data_table.destroy()` This keeps the table structure and only allows for a DataTable when the session is started.